File: tryout/backend/api/src/main/scala/kpn/api/custom/RouteScopeLetter.scala
package kpn.api.custom

import kpn.api.common.RouteScope

object RouteScopeLetter {
  def letter(scope: RouteScope): String = {
    scope match {
      case RouteScope.local => "l"
      case RouteScope.regional => "r"
      case RouteScope.national => "n"
      case RouteScope.international => "i"
      case RouteScope.unknown => "l"
    }
  }
}

File: backend/api/src/main/java/kpn/api/custom/RouteScopeLetter.java
package kpn.api.custom;

import kpn.api.common.RouteScope;

public class RouteScopeLetter {

  public static String letter(RouteScope scope) {
    // Match the scope to determine the letter
    switch (scope) {
      case RouteScope.LOCAL:
        return "l";
      case RouteScope.REGIONAL:
        return "r";
      case RouteScope.NATIONAL:
        return "n";
      case RouteScope.INTERNATIONAL:
        return "i";
      case RouteScope.UNKNOWN:
        return "l";
      default:
        throw new IllegalStateException("Unexpected value: " + scope);
    }
  }

}
